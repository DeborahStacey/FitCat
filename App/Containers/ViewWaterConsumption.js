import React from 'react'
import { Text, ScrollView, View } from 'react-native'
import { default as WellCatManager } from '../Services/WellCatManager'
import I18n from 'react-native-i18n'

import styles from './Styles/ConsumptionStyles'

/*
Name
Brand OPTIONAL
Description OPTIONAL
*/

export default class ViewWaterConsumption extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      waterConsumption: []
    }
  }

  buildViewWaterConsumption = () => {
    var waterRecords = this.state.waterConsumption
    return waterRecords.map((record, index) => {
      var date = record.date
      var consumption = record.waterConsumption + I18n.t('cups_of_water')
      if (record.waterConsumption) {
        return (
          <View key={index}>
            <Text style={styles.labelText}>
              {date}
            </Text>
            <Text style={styles.valueText}>
              {consumption}
            </Text>
          </View>
        )
      }
      return (
        <View key={index}>
          <Text style={styles.labelText}>
            {date}
          </Text>
          <Text style={styles.valueText}>
            No water consumption recorded.
          </Text>
        </View>
      )
    })
  }

  componentWillMount () {
    WellCatManager.getWaterConsumptionHistory().then(response => {
      this.setState({waterConsumption: response})
    })
  }

  render () {
    return (
      <View style={styles.mainContainer}>
        <ScrollView style={styles.container}>
          <View style={styles.paddedContainer}>
            {this.state.waterConsumption && this.buildViewWaterConsumption()}
          </View>
        </ScrollView>
      </View>
    )
  }
}
