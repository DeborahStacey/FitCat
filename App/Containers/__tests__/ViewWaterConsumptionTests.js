import React from 'react'
import ViewWaterConsumption from '../ViewWaterConsumption'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

it('renders the view water consumption page correctly', () => {
  const tree = renderer.create(
    <ViewWaterConsumption />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
