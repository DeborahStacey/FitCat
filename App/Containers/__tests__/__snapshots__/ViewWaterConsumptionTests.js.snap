exports[`test renders the view water consumption page correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F1F5FF",
      "flex": 1,
      "marginTop": 64,
    }
  }>
  <ScrollView
    style={
      Object {
        "flex": 1,
        "paddingTop": 10,
      }
    }>
    <View
      style={
        Object {
          "padding": 25,
        }
      } />
  </ScrollView>
</View>
`;
